src/diffusers/utils/lpw.py

```diff
@@ -0,0 +1,508 @@
+"""
+Implements encode_prompt with long-prompt weighting
+
+adoped from lpw_stable_diffusion by github.com/SkyTNT
+"""
+
+
+import re
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import torch
+from packaging import version
+
+from diffusers.loaders import TextualInversionLoaderMixin, LoraLoaderMixin
+from diffusers.utils import logging
```

Comment:
```suggestion
from ...loaders import TextualInversionLoaderMixin, LoraLoaderMixin
from .. import logging
```

Reply:
I got an error:
    from ...loaders import TextualInversionLoaderMixin, LoraLoaderMixin
ImportError: attempted relative import beyond top-level package

Reply:
another error for logging:
```
from .. import logging
ImportError: cannot import name 'logging' from partially initialized module 'diffusers' (most likely due to a circular import) (/home/imgen/miniconda3/envs/py31/lib/python3.10/site-packages/diffusers/__init__.py)
```